cmd/minikube/cmd/start.go

```diff
@@ -143,6 +147,20 @@ func runStart(cmd *cobra.Command, args []string) {
 		os.Setenv(k, v)
 	}
 
+	registrator, err := registration.DetectRegistrator(host.Driver)
+	if err != nil {
+		fmt.Println("Distribution doesn't support registration")
+	} else {
+		fmt.Printf("Registration with %s...", registrator)
+		m := make(map[string]string)
+		m["username"] = viper.GetString(username)
```

Comment:
You probably need to check here as well whether the parameters are set or not. If you find a registrar and username/password are not specified, you can abort with an appropriate message